### PR TITLE
Bump tar from 6.1.13 to 6.2.1

### PR DESCRIPTION
**Pull request by LDrago27**
### Description

Bumps the tar package from 6.1.13 to 6.2.1. It is a complete version of https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6397 which is linked to CVE(https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6488) mentioned here.

## Changelog
- security: [CVE-2024-28863] Bump tar from 6.1.11 to 6.2.1

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ananzh — APPROVED** [review]

**Comment by AMoo-Miki** [comment]
We need to make sure the package that uses the specific `tar@6.1.11` is not drastically impacted by this higher version.

**Comment by LDrago27** [comment]
Among all the packages that had dependency on **tar**. The package that locked in the version of tar was **geckodriver**. The geckodriver package doesn't lock down the packages in general and uses the tar dependency to just extract the dependency file. 
`if (outFile.indexOf('.tar.gz') >= 0) {
      tar.extract({
        file: archivePath,
        cwd: targetDirectoryPath
      }).then(function (err) {
        if (err) {
          reject(err);
        } else {
          resolve();
        }
      });
    }`
Hence upgrading the tar dependency wouldn't adversely affect the package geckodriver

File: package.json [hunk 1]
```diff
@@ -228,7 +229,7 @@
     "set-value": "^4.1.0",
     "source-map-support": "^0.5.19",
     "symbol-observable": "^1.2.0",
-    "tar": "^6.1.11",
+    "tar": "^6.2.1",
```
Comment (AMoo-Miki) [inline package.json:232]: Do we still need this now that we have `"**/geckodriver/tar": "^6.2.1"`?

**Comment by ananzh** [comment]
Changelog should be `security` not `chore`. 
Can use this one https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6770 as a reference

File: package.json (hunk 1 above)
Comment (ananzh) [reply package.json:232]: I think we should do either 1) "**/tar" or just 2) `"tar": "^6.2.1"`, depending on how tar is used. Don't need to change two places.

Comment (LDrago27) [reply package.json]: Updating the  "**/tar" resolution to utilize the 6.2.1 version.

**Comment by LDrago27** [comment]
> Changelog should be `security` not `chore`. Can use this one #6770 as a reference. It will create a changelog file automatically in the `changelogs/fragments`. You could just remove your changelog.md changes.

Updated the change log

**Review by ananzh — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**